app/javascript/utils/react-bootloader.jsx

```diff
@@ -53,55 +91,62 @@ const render = (elem, component) => {
     </React.StrictMode>,
     elem,
     () => {
-      setTimeout(() => {
-        elem.classList.add('--hydrated')
-      }, 1)
+      // console.log(Date.now(), 'rendered')
+      elem.classList.add('--hydrated')
     }
   )
 
   const unloadOnce = () => {
     ReactDOM.unmountComponentAtNode(elem)
     document.removeEventListener('turbo:before-render', unloadOnce)
   }
-  document.addEventListener('turbo:before-render', unloadOnce)
+  // document.addEventListener('turbo:before-render', unloadOnce)
 }
 
-const renderComponents = (parentElement, mappings) => {
+function renderComponents(parentElement, mappings) {
+  // console.log(Date.now(), 'renderComponents()')
   if (!parentElement) {
     parentElement = document.body
   }
 
-  for (const [name, generator] of Object.entries(mappings)) {
-    const selector = '[data-react-' + name + ']'
-    parentElement.querySelectorAll(selector).forEach((elem) => {
-      const data = JSON.parse(elem.dataset.reactData)
-      render(elem, generator(data, elem))
-    })
+  const elems = Array.from(
+    parentElement.getElementsByClassName('c-react-component')
+  )
+  for (let elem of elems) {
```

Comment:
```suggestion
  for (const elem of parentElement.getElementsByClassName('c-react-component')) {
```

Reply:
```suggestion
  // As getElementsByClassName returns a live collection, it is recommended to use Array.from
  // when iterating through it, otherwise the number of elements may change mid-loop.
  const elems = Array.from(
    parentElement.getElementsByClassName('c-react-component')
  )
  for (let elem of elems) {
```

Reply:
```suggestion
  const elems = Array.from(
    parentElement.getElementsByClassName('c-react-component')
  )
  for (const elem of elems) {
```